Gems/ScriptCanvas/Code/Editor/View/Windows/MainWindow.cpp

```diff
@@ -3873,7 +3873,9 @@ namespace ScriptCanvasEditor
         contextMenu.AddMenuAction(aznew ConvertReferenceToVariableNodeAction(&contextMenu));
         contextMenu.AddMenuAction(aznew ExposeSlotMenuAction(&contextMenu));
         contextMenu.AddMenuAction(aznew CreateAzEventHandlerSlotMenuAction(&contextMenu));
-        contextMenu.AddMenuAction(aznew SetDataSlotTypeMenuAction(&contextMenu));
+
+        // disabling until references can be changed
+        // contextMenu.AddMenuAction(aznew SetDataSlotTypeMenuAction(&contextMenu));
```

Comment:
Rather than disabling this, I recommend seeing it all the way through. One approach may be not to bother with handling the type change, but rather delete/create the variable with the same name but different type in SetDataSlotTypeMenuAction::TriggerAction. Alternatively, if that proves heavy handed, it should be feasible to write a ComvertSlotReferenceType function that is implemented similarly to ConvertslotToReference.

Reply:
Disabled as discussed. The followed through fix will be my next PR.